pkg/tests/mappings/mappings_test.bzl

```diff
@@ -974,3 +975,53 @@ def mappings_unit_tests():
         "mappings_unit_tests",
         strip_prefix_test,
     )
+
+def _gen_manifest_test_main_impl(ctx):
+    ctx.actions.expand_template(
+        template = ctx.file._template,
+        output = ctx.outputs.out,
+        substitutions = {
+            "${EXPECTED}": ctx.attr.expected,
+            "${TARGET}": ctx.attr.target,
+            "${TEST_NAME}": ctx.attr.test_name,
+        },
+    )
+    return [
+        DefaultInfo(files = depset([ctx.outputs.out])),
+    ]
```

Comment:
Since `out` is a predeclared output, isn't this unnecessary?

Reply:
Yes, but I think rules are more readable when they are explicit about their returned providers. Someone can see the implementation and know what the returns are, without having to go to the definition.